### ISSUE TITLE
Routing does not cache by the elastic apm-rum-vue

### ISSUE THREAD

**Issue by Dulajdeshan** [elastic/apm-agent-rum-js#1229]
We have integrated apm client to a vuejs application as below.

```
const router = new VueRouter(
     mode: 'history',
    routes: [
       {path: '/home', component: Home},
        {path: '/dashboard', component: Dashboard},
     ]
)
```
```
Vue.use(ApmVuePlugin, {
  router,
  config: {
    serviceName: 'xxxx',
    serverUrl: 'xxxxxx',
    secretToken: 'xxxx',
    environment: 'xxxx'
  }
})
```

But as we noticed it routing transactions for the direct page visits are only available in the elastic apm. Which means it doesn't catch any internal routing. (As an example if we navigate from '/home' page to '/dashboard' page using a button click, We coudn't see a transaction for that navigation in the elastic apm.

Anyone have an idea about this issue?

**Comment by devcorpio** [comment]
Hi @Dulajdeshan,

Thanks fo reaching out!

> But as we noticed it routing transactions for the direct page visits are only available in the elastic apm.

If I understand properly you are only seeing transactions when accessing pages directly.  And I'm also assuming that the type of those transactions is "page-load" (see image below), is that right? 

<img width="256" alt="Screenshot 2022-05-19 at 12 24 29" src="https://user-images.githubusercontent.com/15065076/169272293-5cc5f22d-2376-4f4c-9386-1da53795137b.png">

> Which means it doesn't catch any internal routing. (As an example if we navigate from '/home' page to '/dashboard' page using a button click, We coudn't see a transaction for that navigation in the elastic apm.

You may need a XHR/fetch within a route change to make the transaction to be finished and sent to the server. It is important to bear in mind that transactions need to have at least one span (like the ones related to XHR/Fetch requests), otherwise the agent discards them.

See this [link](https://github.com/elastic/apm-agent-rum-js/issues/1163#issuecomment-1034785771) for more detailed information

Let me know if this helps you

Thanks,
Alberto

**Comment by devcorpio** [comment]
Hi @Dulajdeshan,

Have you been able to solve the routing issue?

Thanks,
Alberto

**Comment by Dulajdeshan** [comment]
@devcorpio Yes, problem was there should be at least one span for a transaction. I have added a default span for all these routing transactions.
